pyteal/ast/itxn_test.py

```diff
@@ -128,7 +138,7 @@ def test_InnerTxnBuilder_SetFields():
         actual.addIncoming()
         actual = pt.TealBlock.NormalizeBlocks(actual)
 
-        with pt.TealComponent.Context.ignoreExprEquality():
+        with pt.TealComponent.Context.ignoreScratchSlotEquality(), pt.TealComponent.Context.ignoreExprEquality():
```

Comment:
If you compare with `TealComponent.Context.ignoreScratchSlotEquality()`, please follow with a call to `TealBlock.MatchScratchSlotReferences`.

Example: https://github.com/algorand/pyteal/blob/79df68ba0d02fd56c242cdaaf84929fe7908b003/pyteal/ast/abi/array_base_test.py#L157-L160